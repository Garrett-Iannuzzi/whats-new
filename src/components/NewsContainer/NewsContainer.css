.article-section {
  display: flex;
  flex-direction: column;
  flex-flow: wrap;
}