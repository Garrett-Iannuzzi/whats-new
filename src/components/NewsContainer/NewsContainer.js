import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'

const  NewsContainer = ({ articles }) => {
  const newsArticles = articles.map(article => {
    return (
      <NewsArticle
        key={article.id}
        title={article.title}
        img={article.img}
        description={article.description}
        url={article.url}
      />
    )
  })

  return (
    <section className="article">
      {newsArticles}
    </section>
  )

}
export default NewsContainer;