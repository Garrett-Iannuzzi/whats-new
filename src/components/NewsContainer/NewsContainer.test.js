import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('News Container', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    let articles = [
      { img: '//www.img.com', headline: 'Go Nuggets', description: 'They win 5th' }, 
      { img: '//www.img.com', headline: 'Go Broncos', description: 'Spoiler, they suck'}
    ]
    const wrapper = shallow(<NewsContainer 
      articles={articles}
    />)
    expect(wrapper).toMatchSnapshot();
  });

})