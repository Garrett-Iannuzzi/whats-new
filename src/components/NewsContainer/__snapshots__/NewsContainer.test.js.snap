// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Container should match the snapshot with all data passed in correctly 1`] = `
<section
  className="article-section"
>
  <NewsArticle
    description="They win 5th"
    img="//www.img.com"
    title="Go Nuggets"
  />
  <NewsArticle
    description="Spoiler, they suck"
    img="//www.img.com"
    title="Go Broncos"
  />
</section>
`;
